### ISSUE TITLE
global: record receivers should be centralized

### ISSUE THREAD

**Issue by jacquerie** [inspirehep/inspire-next#1410]
We currently have record receivers in the following files:

```
inspirehep/dojson/hep/receivers.py
inspirehep/modules/authors/receivers.py
inspirehep/modules/orcid/receivers.py
inspirehep/modules/records/receivers.py
```

and soon in

```
inspirehep/modules/disambiguation/receivers.py
```

The problem is that record receivers are **global**, and run no matter what the type of the record (HEP, Author...) is. Reasoning about them is already pretty difficult, and having them scattered all over the codebase will become a manteinance nightmare.

So I propose to centralize all of them in `inspirehep/modules/records/receivers.py`, perhaps with some decorators that restrict the kind of record that receiver should fire for.

This is currently blocked by the merging of #1353 and #1408.

- [x] Merge `inspirehep/dojson/hep/receivers.py` (#1841)
- [x] Merge `inspirehep/modules/authors/receivers.py`
- [ ] Merge `inspirehep/modules/disambiguation/receivers.py`
- [ ] Merge `inspirehep/modules/orcid/receivers.py`
- [ ] Merge `inspirehep/modules/workflows/receivers.py`

**Comment by jacquerie** [comment]
Another issue with having receivers spread out in the codebase: they are added upon import, so *maybe* an application (like a CLI, or a debug session) can end up having or not having some receivers loaded, leading to all sorts of nasty bugs.

**Comment by jacquerie** [comment]
This happened in https://github.com/inspirehep/inspire-next/pull/2766 and https://github.com/inspirehep/inspire-next/pull/2754.
